File: VB/VerticalGridGroupHotTrack/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraVerticalGrid.Events
Imports System.Drawing
Imports DevExpress.XtraVerticalGrid.Rows
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.Utils
Imports DevExpress.Utils.Paint
Imports DevExpress.XtraVerticalGrid.ViewInfo

Namespace VerticalGridGroupHotTrack
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
			Me.productsTableAdapter.Fill(Me.nwindDataSet.Products)

		End Sub

		Private Sub OnCustomDrawRowHeaderCell(ByVal sender As Object, ByVal e As CustomDrawRowHeaderCellEventArgs) Handles verticalGrid.CustomDrawRowHeaderCell
			Dim hitPoint As Point = MousePosition
			'if (!(e.Row is CategoryRow)) return;
			Dim grid As VGridControl = CType(sender, VGridControl)
			Dim hitInfo As VGridHitInfo = (CType(sender, VGridControl)).CalcHitInfo(grid.PointToClient(hitPoint))
			If hitInfo.Row IsNot e.Row Then
				Return
			End If
			Dim rowInfo As BaseRowViewInfo = Nothing
			For Each info As BaseRowViewInfo In grid.ViewInfo.RowsViewInfo
				If TypeOf info Is CategoryRowViewInfo Then
					Continue For
				End If
				rowInfo = info
				Exit For
			Next info
			If rowInfo Is Nothing Then
				Return
			End If
			Dim valueInfo As RowValueInfo = Nothing
			For Each info As RowValueInfo In rowInfo.ValuesInfo
				If info.Bounds.X < hitInfo.PtMouse.X AndAlso info.Bounds.Right > hitInfo.PtMouse.X Then
					valueInfo = info
					Exit For
				End If
			Next info
			If valueInfo Is Nothing Then
				Return
			End If
			e.Handled = True
			e.Appearance.DrawBackground(e.Cache, e.Bounds)
			e.Appearance.DrawString(e.Cache, e.Caption, e.CaptionRect)
			If e.ImageIndex > 0 Then
				ImageCollection.DrawImageListImage(e.Cache, e.Row.Properties.Images, e.ImageIndex, e.ImageRect)
			End If
			If e.Focused Then
				XPaint.Graphics.DrawFocusRectangle(e.Graphics, e.FocusRect, e.Appearance.ForeColor, e.Appearance.BackColor)
			End If
			Dim hotTrackRect As New Rectangle(valueInfo.Bounds.X, e.Bounds.Y, valueInfo.Bounds.Width, e.Bounds.Height)
			hotTrackRect.Inflate(-1, -1)
			e.Graphics.FillRectangle(Brushes.AliceBlue, hotTrackRect)
			e.Graphics.DrawRectangle(Pens.Blue, hotTrackRect)
		End Sub

		Private Sub OnVGridControlMouseMove(ByVal sender As Object, ByVal e As MouseEventArgs) Handles verticalGrid.MouseMove
			CType(sender, VGridControl).Invalidate()
		End Sub
	End Class
End Namespace

File: CS/VerticalGridGroupHotTrack/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraVerticalGrid.Events;
using System.Drawing;
using DevExpress.XtraVerticalGrid.Rows;
using DevExpress.XtraVerticalGrid;
using DevExpress.Utils;
using DevExpress.Utils.Paint;
using DevExpress.XtraVerticalGrid.ViewInfo;

namespace VerticalGridGroupHotTrack {
    public partial class Form1 :Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
            this.productsTableAdapter.Fill(this.nwindDataSet.Products);

        }

        private void OnCustomDrawRowHeaderCell(object sender, CustomDrawRowHeaderCellEventArgs e) {
            Point hitPoint = MousePosition;
            //if (!(e.Row is CategoryRow)) return;
            VGridControl grid = (VGridControl)sender;
            VGridHitInfo hitInfo = ((VGridControl)sender).CalcHitInfo(grid.PointToClient(hitPoint));
            if (hitInfo.Row != e.Row) return;
            BaseRowViewInfo rowInfo = null;
            foreach (BaseRowViewInfo info in grid.ViewInfo.RowsViewInfo) {
                if (info is CategoryRowViewInfo) continue;
                rowInfo = info;
                break;
            }
            if (rowInfo == null) return;
            RowValueInfo valueInfo = null;
            foreach (RowValueInfo info in rowInfo.ValuesInfo)
                if (info.Bounds.X < hitInfo.PtMouse.X &&
                    info.Bounds.Right > hitInfo.PtMouse.X) {
                    valueInfo = info;
                    break;
                }
            if (valueInfo == null) return;
            e.Handled = true;
            e.Appearance.DrawBackground(e.Cache, e.Bounds);
            e.Appearance.DrawString(e.Cache, e.Caption, e.CaptionRect);
            if (e.ImageIndex > 0)
                ImageCollection.DrawImageListImage(e.Cache, e.Row.Properties.Images, e.ImageIndex,
                    e.ImageRect);
            if (e.Focused)
                XPaint.Graphics.DrawFocusRectangle(e.Graphics, e.FocusRect, e.Appearance.ForeColor,
                    e.Appearance.BackColor);
            Rectangle hotTrackRect = new Rectangle(valueInfo.Bounds.X, e.Bounds.Y,
                valueInfo.Bounds.Width, e.Bounds.Height);
            hotTrackRect.Inflate(-1, -1);
            e.Graphics.FillRectangle(Brushes.AliceBlue, hotTrackRect);
            e.Graphics.DrawRectangle(Pens.Blue, hotTrackRect);
        }

        private void OnVGridControlMouseMove(object sender, MouseEventArgs e) {
            ((VGridControl)sender).Invalidate();
        }
    }
}